web/src/app/administration/components/workbasket-list-toolbar/workbasket-list-toolbar.component.spec.ts

```diff
@@ -0,0 +1,118 @@
+import { CreateWorkbasket } from '../../../shared/store/workbasket-store/workbasket.actions';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { Filter } from '../../../shared/models/filter';
+import { Sorting } from '../../../shared/models/sorting';
+import { ACTION } from '../../../shared/models/action';
+import { TaskanaType } from '../../../shared/models/taskana-type';
+
+const getDomainFn = jest.fn().mockReturnValue(true);
+const domainServiceMock = jest.fn().mockImplementation(
+  (): Partial<DomainService> => ({
+    getDomains: getDomainFn
+  })
+);
+
+@Component({ selector: 'taskana-administration-import-export', template: '' })
+class ImportExportStub {
+  @Input() currentSelection: TaskanaType;
+}
+
+@Component({ selector: 'taskana-shared-sort', template: '' })
+class SortStub {
+  @Input() sortingFields: Map<string, string>;
+  @Input() defaultSortBy = 'key';
+  @Output() performSorting = new EventEmitter<Sorting>();
+}
+
+@Component({ selector: 'taskana-shared-filter', template: '' })
+class FilterStub {
+  @Output() performFilter = new EventEmitter<Filter>();
+}
+
+describe('WorkbasketListToolbarComponent', () => {
+  let fixture: ComponentFixture<WorkbasketListToolbarComponent>;
+  let debugElement: DebugElement;
+  let component: WorkbasketListToolbarComponent;
+  let store: Store;
+  let actions$: Observable<any>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule,
+        NgxsModule.forRoot([WorkbasketState]),
+        MatSnackBarModule,
+        MatDialogModule,
+        BrowserAnimationsModule
+      ],
+      declarations: [WorkbasketListToolbarComponent, ImportExportStub, SortStub, FilterStub],
+      providers: [{ provide: DomainService, useClass: domainServiceMock }, WorkbasketService]
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(WorkbasketListToolbarComponent);
+    debugElement = fixture.debugElement;
+    component = fixture.debugElement.componentInstance;
+    store = TestBed.inject(Store);
+    actions$ = TestBed.inject(Actions);
+    component.action = ACTION.COPY;
+  }));
+
+  it('should create the component', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should dispatch CreateWorkbasket when addWorkbasket is called', async((done) => {
+    component.addWorkbasket();
+    let actionDispatched = false;
+    actions$.pipe(ofActionDispatched(CreateWorkbasket)).subscribe(() => {
+      actionDispatched = true;
+      expect(actionDispatched).toBe(true);
+      done();
+    });
+  }));
+
+  it('should not do anything in addWorkbasket if action is create ', async((done) => {
+    component.action = ACTION.CREATE;
+    fixture.detectChanges();
+    component.addWorkbasket();
+    let actionDispatched = false;
+    actions$.pipe(ofActionDispatched(CreateWorkbasket)).subscribe(() => {
+      actionDispatched = true;
+      expect(actionDispatched).toBe(false);
```

Comment:
moved expect out of subscription, this subscription shouldn't be called so that actionDispatched should always be false. Expect false at the end of test case will still be checked